packages/sign-client/src/controllers/engine.ts

```diff
@@ -1829,23 +1832,42 @@ export class Engine extends IEngine {
     topic,
     payload,
   ) => {
-    const { requester, authPayload, expiryTimestamp } = payload.params;
-    const hash = hashMessage(JSON.stringify(payload));
-    const verifyContext = await this.getVerifyContext(hash, this.client.metadata);
-    const pendingRequest = {
-      requester,
-      pairingTopic: topic,
-      id: payload.id,
-      authPayload,
-      verifyContext,
-      expiryTimestamp,
-    };
-    await this.setAuthRequest(payload.id, { request: pendingRequest, pairingTopic: topic });
-    this.client.events.emit("session_authenticate", {
-      topic,
-      params: payload.params,
-      id: payload.id,
-    });
+    try {
+      const { requester, authPayload, expiryTimestamp } = payload.params;
+      const hash = hashMessage(JSON.stringify(payload));
+      const verifyContext = await this.getVerifyContext(hash, this.client.metadata);
+      const pendingRequest = {
+        requester,
+        pairingTopic: topic,
+        id: payload.id,
+        authPayload,
+        verifyContext,
+        expiryTimestamp,
+      };
+      await this.setAuthRequest(payload.id, { request: pendingRequest, pairingTopic: topic });
+      this.client.events.emit("session_authenticate", {
+        topic,
+        params: payload.params,
+        id: payload.id,
+      });
+    } catch (err: any) {
+      const receiverPublicKey = payload.params.requester.publicKey;
+      const senderPublicKey = await this.client.core.crypto.generateKeyPair();
+
+      const encodeOpts = {
+        type: TYPE_1,
+        receiverPublicKey,
+        senderPublicKey,
+      };
+      await this.sendError({
+        id: payload.id,
+        topic,
+        error: err,
+        encodeOpts,
+        rpcOpts: ENGINE_RPC_OPTS.wc_sessionAuthenticate.autoReject,
+      });
+      this.client.logger.error(err);
```

Comment:
I'd suggest to move this line to be the first one line in the catch block. It should be easier to debug when something goes wrong.